### PR TITLE
Set default genesis in mobile client from network ID

### PR DESCRIPTION
**Pull request by nategraf**
### Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit 4493a3e7a9e028368c023e5ca3fc5e52bff3db1b
<details><summary>coverage:  51.0% of statements across all listed packages</summary><br>
<pre>
coverage:  66.1% of statements in consensus/istanbul
coverage:  41.3% of statements in consensus/istanbul/announce
coverage:  54.6% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  70.4% of statements in consensus/istanbul/core
coverage:  45.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  75.3% of statements in consensus/istanbul/uptime
coverage: 100.0% of statements in consensus/istanbul/uptime/store
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 901e9401c1
</details>

File: mobile/geth.go [hunk 1]
```diff
@@ -243,7 +247,12 @@ func NewNode(datadir string, config *NodeConfig) (stack *Node, _ error) {
 				config.EthereumNetworkID = int64(params.BaklavaNetworkId)
 			}
 		}
+	} else if config.EthereumNetworkID == int64(params.AlfajoresNetworkId) {
+		genesis = core.DefaultAlfajoresGenesisBlock()
+	} else if config.EthereumNetworkID == int64(params.BaklavaNetworkId) {
+		genesis = core.DefaultBaklavaGenesisBlock()
 	}
```
Comment (jeanregisser) [inline mobile/geth.go:254]: Great! Just to be sure, we don't need a case for the `MainnetNetworkId` because it's already the default?

**Comment by jeanregisser** [comment]
And one more question, what about the backward compatibility for existing Valora users who had been initializing the node by passing the genesis data?
Is there some persisted data somewhere down the line that would create a problem by switching from passing the genesis data to not passing it anymore? Especially related to the Espresso hardfork.

**Comment by nategraf** [comment]
> And one more question, what about the backward compatibility for existing Valora users who had been initializing the node by passing the genesis data? Is there some persisted data somewhere down the line that would create a problem by switching from passing the genesis data to not passing it anymore? Especially related to the Espresso hardfork.

If we additionally change Valora to not pass in the chain config parameters, I don't think there should be an issue. This is certainly something we should test.

File: mobile/geth.go (hunk 1 above)
Comment (nategraf) [reply mobile/geth.go]: Default was to use mainnet, but I added a case to make it more explicit.

**Comment by nategraf** [comment]
Following the process for making releases, this PR should be merged to `master` first, then cherry picked back to `release/1.5.x`

**Pull request closed without merging**